src/cursor/cur_file.c

```diff
@@ -194,6 +288,7 @@ __curfile_search(WT_CURSOR *cursor)
 
     cbt = (WT_CURSOR_BTREE *)cursor;
     CURSOR_API_CALL(cursor, session, search, CUR2BT(cbt));
+    F_CLR(cbt, WT_CBT_RESEARCH);
```

Comment:
Hadn't thought about clearing that flag -- it seems we'll need to clear it on all methods? I hate to rely on doing that everywhere explicitly, but I don't have a better idea. We could mess with the `api.h` macros, but I'm not sure if that would be better or not.